enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java

```diff
@@ -102,9 +104,19 @@ public JWTAuthenticator() {
 
     @Override
     public boolean canAuthenticate(RequestContext requestContext) {
+        String apiType = requestContext.getMatchedAPI().getApiType();
         if (isJWTEnabled(requestContext)) {
             String jwt = retrieveAuthHeaderValue(requestContext);
-            return jwt != null && jwt.split("\\.").length == 3;
+
+            if (apiType.equalsIgnoreCase(APIConstants.ApiType.WEB_SOCKET)) {
+                if (jwt == null) {
+                    jwt = extractJWTInWSProtocolHeader(requestContext);
+                }
+                AuthenticatorUtils.addWSProtocolResponseHeaderIfRequired(requestContext,
+                        Constants.WS_OAUTH2_KEY_IDENTIFIED);
+
+                return jwt != null && jwt.split("\\.").length == 3;
```

Comment:
shouldn't this return be outside of the if block?

Reply:
yes